tests/formik/ActionBlock.test.js

```diff
@@ -94,13 +91,43 @@ describe("formik/ActionBlock", () => {
     );
   });
 
-  it("should reset the form fields when cancel button is clicked", () => {
+  it("should reset the form fields when cancel button is clicked", async () => {
     render(<TestActionBlock />);
 
     const cancelButton = screen.getByRole("button", { name: /Cancel/i });
     const input = screen.getByRole("textbox");
     userEvent.type(input, "test");
     userEvent.click(cancelButton);
-    expect(input).toHaveValue("");
+    await waitFor(() => expect(input).toHaveValue(""));
+  });
+
+  it("Cancel button should be disabled if form value is same", async () => {
+    const onSubmit = jest.fn();
+    render(<TestActionBlock input="Oliver" onSubmit={onSubmit} />);
+
+    const cancelButton = screen.getByRole("button", { name: /Cancel/i });
+    expect(cancelButton).toBeDisabled();
+    const input = screen.getByRole("textbox");
+    userEvent.type(input, "test");
+    await waitFor(() => expect(cancelButton).not.toBeDisabled());
+    userEvent.type(input, repeat("{backspace}", "test".length).join(""));
+    await waitFor(() => expect(cancelButton).toBeDisabled());
+    userEvent.click(cancelButton);
+    await waitFor(() => expect(onSubmit).not.toBeCalled());
+  });
+
+  it("Cancel button should be disabled when submitting", async () => {
+    const onSubmit = jest.fn();
+    render(<TestActionBlock input="Oliver" onSubmit={new Promise(() => {})} />);
+
+    const cancelButton = screen.getByRole("button", { name: /Cancel/i });
+    expect(cancelButton).toBeDisabled();
+    const input = screen.getByRole("textbox");
+    userEvent.type(input, "test");
+    await waitFor(() => expect(cancelButton).not.toBeDisabled());
+    userEvent.type(input, repeat("{backspace}", "test".length).join(""));
+    await waitFor(() => expect(cancelButton).toBeDisabled());
```

Comment:
We don't have to check this condition in the case for submitting right?

Reply:
When form is being submitted, cancel button should be disabled.
There is no point in reverting the changes while the form is being submitted because immediately after the API succeeds, the form value will be overwritten by the latest changes.

For a user, the cancel button means something else during form submission. They might expect that the cancel button will interrupt the submission, and the changes won't be saved to the backend, which is not what we actually do with that cancel button.

For both of these reasons, it is best to disable cancel button during submission.

Reply:
@Amaljith-K _a Please update the tests as discussed over call.